### ISSUE TITLE
Using different GNN for the NMT task

### ISSUE THREAD

**Issue by smith-co** [graph4ai/graph4nlp#530]
## ❓ Questions and Help

I want to use different GNN architecture for the NMT model that is provided. I presume I set the "graph_embedding_name:" to different GNN architectures, i.e.:

- gat 
- graphsage
- ggnn   

And then adjust the `graph_construction_private` values and that would do?

I am particularly interested to run NMT with GAT as that makes more sense as I can use copy mechanism. Of course for my case the input and output has same vocab. Do I have to make any fundamental change to the NMT example already provided? Or would that work as is?

**Comment by AlanSwift** [comment]
You can change GNN to gat, graphsage and ggnn according to your need.

For copy, you have to change the code. Please refer to the [semantic_parsing](https://github.com/graph4ai/graph4nlp/tree/master/examples/pytorch/semantic_parsing/graph2seq), QG, summarization's implementation.

**Comment by smith-co** [comment]
@AlanSwift For NMT, `GAT with copy mechanism` is very common. All the NMT model like OpenNMT support it. Do you plan to add an example for that, please🙏?

**Comment by smith-co** [comment]
Regarding the number of heads, do you recommend any heuristics regarding how many heads to use?

```
  graph_embedding_private:
    heads: [2, 2, 1]
    attn_drop: 0.2
    negative_slope: 0.2
    residual: true
    activation: "relu"
    allow_zero_in_degree: true
```

**Comment by AlanSwift** [comment]
1. We are sorry that the copy for NMT is not on my plan currently. I will do it in the future month if this is a need.
There are some suggestions: #537 .
2. Empirically, 2-4 is worth trying. However, I strongly recommend trying different parameters in specific applications.

**Comment by smith-co** [comment]
@AlanSwift for 2. are you referring to use number of `heads` between 2-4.?

**Comment by smith-co** [comment]
@AlanSwift what do you suggest for the NMT task?

- can you please point to me what parameters are you referring for the number of heads with GAT? 
- Also what number of training steps what you recommend to try with?

**Comment by AlanSwift** [comment]
1. 3-6 is for most conditions.
2. This is hard to tell. Please set some metric monitors.
